### ISSUE TITLE
BUG: Cannot read properties of undefined (reading 'importedBy')

### ISSUE THREAD

**Issue by jermsam** [Menci/vite-plugin-top-level-await#32]
Error thrown by 

         moduleInfo.imported.forEach(importedModuleName => {
            bundleInfo[importedModuleName].importedBy.push(moduleName);
        });

in bundle.info.js
and  a similar one by
        
       
      if (bundleInfo[importedModuleName].transformNeeded) {
            importDeclaration.specifiers.push((0, make_node_1.makeImportSpecifier)(options.promiseExportName, 
             options.promiseImportName(importedPromiseCount)));
            importedPromiseCount++;
        }
        
in transform.js
        
Escaping these with ? fixes the bug for me:
        
         moduleInfo.imported.forEach(importedModuleName => {
            bundleInfo[importedModuleName]?.importedBy.push(moduleName);
        });
        
and 
        

        if (bundleInfo[importedModuleName]?.transformNeeded) {
            importDeclaration.specifiers.push((0, make_node_1.makeImportSpecifier)(options.promiseExportName, 
            options.promiseImportName(importedPromiseCount)));
            importedPromiseCount++;
        }

**Comment by Menci** [comment]
Could you please provide a simple project to reproduce it? I'm not sure whether this could be ignored by adding a `?` or not. Maybe I wrongly resolved the import path and caused the incorrect detection of TLA usage.

**Comment by jermsam** [comment]
Any simple implementation will throw the error when you build
E.g: At: https://stackblitz.com/edit/vitejs-vite-xihreo?file=vite.config.ts,package.json&terminal=dev
yarn build

**Comment by Menci** [comment]
I can't reproduce @jermsam

**Comment by abnerlee** [comment]
I'm facing this when using `storybook build`, normal vite dev or build succeeded, however

**Comment by jorbuedo** [comment]
Same, failing for storybook build

**Comment by qisoft** [comment]
Here is a reproduction repo: https://github.com/qisoft/react-vite-federation-top-level-await-bug
Run `pnpm install` and then `pnpm run build:remotes`

**Comment by Menci** [comment]
@qisoft I tested your configuration. Even without my plugin, it generatesthe  bundle with imports to a non-existing module `./__federation_shared_react.js`. So we can't handle it.

```
$ pnpm run build:remotes

> react-vite@1.0.0 build:remotes /Users/Menci/react-vite-federation-top-level-await-bug
> pnpm --parallel --filter "./remote" build

remote build$ vite build
remote build: vite v4.0.3 building for production...
remote build: transforming...
remote build: ✓ 40 modules transformed.
remote build: rendering chunks...
remote build: computing gzip size...
remote build: dist/index.html                                      0.81 kB
remote build: dist/assets/react-35ef61ed.svg                       4.13 kB
remote build: dist/assets/style-998defbc.css                       1.97 kB │ gzip: 0.86 kB
remote build: dist/assets/index-5133fde1.js                        1.67 kB │ gzip: 0.64 kB
remote build: dist/assets/__federation_expose_Button-6ca047b1.js   1.81 kB │ gzip: 0.90 kB
remote build: dist/assets/remoteEntry.js                           1.85 kB │ gzip: 0.78 kB
remote build: dist/assets/__federation_fn_import.js               13.15 kB │ gzip: 2.80 kB
remote build: Done
$ head -n 2 remote/dist/assets/__federation_expose_Button-*.js
import { importShared } from './__federation_fn_import.js';
import { r as reactExports } from './__federation_shared_react.js';
$ find remote/dist | grep react
remote/dist/assets/react-35ef61ed.svg
```

I have no idea of it.

**Comment by 7michele7** [comment]
I can repro the same issue when building storybook:

<img width="1059" alt="Screenshot 2023-08-31 at 17 09 03" src="https://github.com/Menci/vite-plugin-top-level-await/assets/17926167/e3ef00fa-109d-452f-a07a-331c1b843b26">

**Comment by Timur00Kh** [comment]
Same problem when i use  `vite-plugin-top-level-await` with `vite-plugin-lib-assets`.
Reordering plugins doesn't help

**Comment by Menci** [comment]
Closed by c43ba2b121700da52ca31591fc292e3b67f7ce00
